core/Db/TransactionLevel.php

```diff
@@ -40,6 +40,10 @@ public function canLikelySetTransactionLevel()
 
     public function setUncommitted()
     {
+        if ($this->db->supportsUncommitted) {
+            return true;
+        }
```

Comment:
That doesn't seem to make sense to me. Doesn't that mean, that if setting the transaction level isn't supported, that it will be retried every time it's requested. And on the other side - if it's supported the transaction level won't be set again once it was detected.